packages/@o3r/schematics/src/utility/package-manager-runner.ts

```diff
@@ -1,39 +1,64 @@
 import * as commentJson from 'comment-json';
+import { WorkspaceSchema } from '../interfaces';
 
 /** Npm packageManager */
 export type SupportedPackageManagers = 'npm' | 'yarn';
 
+/**
+ * Determine if the given packager manager is supported
+ *
+ * @param name Name of the package manager
+ */
+export function isSupportedPackageManager(name?: any): name is SupportedPackageManagers {
+  return name === 'yarn' || name === 'npm';
+}
+
 /**
  * Get the Package Manager
  *
  * @param enforcedNpmManager package manager to enforce
  */
-export function getPackageManagerName(enforcedNpmManager?: SupportedPackageManagers | '' | undefined | null): SupportedPackageManagers {
-  return enforcedNpmManager || (process.env?.npm_execpath?.includes('yarn') ? 'yarn' : 'npm');
+function getPackageManagerName(enforcedNpmManager?: SupportedPackageManagers): SupportedPackageManagers {
+  const envCustomPackageManager = isSupportedPackageManager(process.env?.ENFORCED_PACKAGE_MANAGER) && process.env.ENFORCED_PACKAGE_MANAGER;
+  const envNodePackageManager = process.env?.npm_execpath?.includes('yarn') && 'yarn';
+  return enforcedNpmManager || envCustomPackageManager || envNodePackageManager || 'npm';
 }
 
 /**
  * Get package manager used in runs
  * Defaults to the package manager setup in process.env if no package manager set in angular.json
  *
- * @param angularJsonString Content of angular.json file
+ * @param options.workspaceConfig Workspace configuration
+ * @param options.enforcedNpmManager  package manager to enforce, will be used if supported
+ * @param options
  */
-export function getPackageManager(angularJsonString?: string | null) {
-  let packageManager = process.env && process.env.npm_execpath && process.env.npm_execpath.indexOf('yarn') === -1 ? 'npm' : 'yarn';
-  if (angularJsonString) {
-    const angularJsonObj = commentJson.parse(angularJsonString) as any;
+export function getPackageManager(options?: {workspaceConfig?: WorkspaceSchema | string | null; enforcedNpmManager?: string | null}) {
```

Comment:
this is a bit breaking. Maybe rename it and deprecate the other.
`getUsedPackageManager` maybe ?

Reply:
Not breaking if committed before 9.0.0

Reply:
ok :) I did not check when was added :)